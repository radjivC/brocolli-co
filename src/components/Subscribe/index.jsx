import React, {Component} from "react"
import Form from 'react-validation/build/form'
import Input from 'react-validation/build/input'
import Button from 'react-validation/build/button'
import { required, email, miniChar } from '../../helpers/validator'
import SubscribeService from '../../services/Subscribe'

export default class Subscribe extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      confirmEmail: '',
      email: '',
      isDone:false,
      error: false
    }
  }

  onSubmit = (e) => {
    e.preventDefault()
    e.target.disabled = true
    if(this.state.email !== this.state.confirmEmail){
      this.setState({error:true})
    }

    SubscribeService.post(this.state)
      .then(data => this.setState({ isDone: true  }))
      .catch(err => this.setState({ error: true  }))
  }

  change = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  Done = () => (
    <div>
      <center>
        <h1>All Done</h1>
      <br />
      <p>You will be one of the first to experience Brocoli and co, when we launch</p>
      <br />
      <button className="button is-success is-medium" onClick={e => this.props.closeModalRequest()}>Done</button>
      </center>
    </div>
  )

  SubscribeForm = () => (
    <Form>
      <div className="is-uppercase">
        <center>
          <h1 className="has-text-info">Request an Invite</h1>
        </center>
        <br />
      </div>
      <div className="field">
        <div className="control has-icons-left">
          <Input className="input" name="name" type="text" placeholder="Name" value={this.state.name} validations={[required, miniChar]} onChange={e => this.change(e)} />
          <span className="icon is-small is-left">
            <i className="fa fa-user" />
          </span>
        </div>
      </div>

      <div className="field">
        <div className="control has-icons-left">
          <Input className="input" name="email" type="email" placeholder="Email" value={this.state.email} validations={[required, email]} onChange={e => this.change(e)} />
          <span className="icon is-small is-left">
            <i className="fa fa-envelope" />
          </span>
        </div>
      </div>

      <div className="field">
        <div className="control has-icons-left">
          <Input className="input" name="confirmEmail" type="confirmEmail" placeholder="Confirm Email" value={this.state.confirmEmail} validations={[required, email]} onChange={e => this.change(e)} />
          <span className="icon is-small is-left">
            <i className="fa fa-envelope" />
          </span>
        </div>
      </div>
      <br />
      <div>
        {this.state.error ? <center><p className="has-text-danger">Error, please try again</p></center> : <center><p>Please enter the details above and then click submit.</p></center> }
        <br />
        <center><Button className="button is-primary is-medium" onClick={e => this.onSubmit(e)}>Submit</Button></center>
      </div>
    </Form>
  )


  render() {
    return (
      <div>
        { this.state.isDone ? this.Done() : this.SubscribeForm() }
      </div>
    )
  }
}
