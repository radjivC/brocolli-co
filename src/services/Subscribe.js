import fetch from 'fetchival'

export class Subscribe {
  async post(user) {
    this.params = {
      email: user.email,
      name: user.name
    }

    const url = `https://l94wc2001h.execute-api.ap-southeast-2.amazonaws.com/prod/fake-auth`
    const data = await fetch(url).post(this.params)
    return data
  }
}

export default new Subscribe()
