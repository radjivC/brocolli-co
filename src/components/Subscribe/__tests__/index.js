import React from 'react'
import { mount } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import Subscribe from '../'

configure({ adapter: new Adapter() })

describe('Subscribe', () => {
  it('can render without crashing', () => {
    const component = renderer.create(
      <Subscribe />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('can render form', () => {
    const wrapper = mount(<Subscribe />)
    wrapper.setState({ isDone:false })
    expect(wrapper.find('input').length).toEqual(3)
    expect(wrapper.find('h1').text()).toEqual('Request an Invite')
    expect(wrapper.find('button.button.is-primary.is-medium').text()).toEqual('Submit')
    expect(wrapper.find('p').text()).toEqual('Please enter the details above and then click submit.')
  })

  it('can render done', () => {
    const wrapper = mount(<Subscribe />)
    wrapper.setState({ isDone:true })
    expect(wrapper.find('input').length).toEqual(0)
    expect(wrapper.find('h1').text()).toEqual('All Done')
    expect(wrapper.find('p').text()).toEqual('You will be one of the first to experience Brocoli and co, when we launch')
    expect(wrapper.find('button.button.is-success.is-medium').text()).toEqual('Done')
  })
})
