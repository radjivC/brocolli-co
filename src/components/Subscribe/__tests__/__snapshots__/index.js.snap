// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe can render without crashing 1`] = `
<div>
  <form>
    <div
      className="is-uppercase"
    >
      <center>
        <h1
          className="has-text-info"
        >
          Request an Invite
        </h1>
      </center>
      <br />
    </div>
    <div
      className="field"
    >
      <div
        className="control has-icons-left"
      >
        <div>
          <input
            className="input"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <span
          className="icon is-small is-left"
        >
          <i
            className="fa fa-user"
          />
        </span>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control has-icons-left"
      >
        <div>
          <input
            className="input"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
        <span
          className="icon is-small is-left"
        >
          <i
            className="fa fa-envelope"
          />
        </span>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control has-icons-left"
      >
        <div>
          <input
            className="input"
            name="confirmEmail"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Confirm Email"
            type="confirmEmail"
            value=""
          />
        </div>
        <span
          className="icon is-small is-left"
        >
          <i
            className="fa fa-envelope"
          />
        </span>
      </div>
    </div>
    <br />
    <div>
      <center>
        <p>
          Please enter the details above and then click submit.
        </p>
      </center>
      <br />
      <center>
        <button
          className="button is-primary is-medium"
          disabled={true}
          onClick={[Function]}
        >
          Submit
        </button>
      </center>
    </div>
  </form>
</div>
`;
